bindings/wallet-cordova/scripts/run_tests.py

```diff
@@ -0,0 +1,145 @@
+        subprocess.check_call(["cordova", "requirements", "android"], cwd=app_dir)
+
+    if ios:
+        subprocess.check_call(["cordova", "platform", "add", "ios"], cwd=app_dir)
+        subprocess.check_call(["cordova", "requirements", "ios"], cwd=app_dir)
+
+
+def install_main_plugin(
+    app_dir: Path, reinstall=True, android=False, ios=False, cargo_build=True
+):
+    if reinstall:
+        subprocess.check_call(
+            ["cordova", "plugin", "rm", "wallet-cordova-plugin"], cwd=app_dir
+        )
+
+    if android:
+        if cargo_build:
+            build_jni(release=False)
+        else:
+            copy_jni_libs(release=False)
+            copy_jni_definitions()
+
+    if ios:
+        build_ios(release=False)
+
+    subprocess.check_call(
+        ["cordova", "plugin", "add", str(plugin_directory)], cwd=app_dir
+    )
+
+
+def install_test_plugin(app_dir: Path, reinstall=True):
+    subprocess.check_call(["npm", "install"], cwd=tests_directory)
+    subprocess.check_call(["npm", "run", "build"], cwd=tests_directory)
+
+    if reinstall:
+        subprocess.check_call(
+            ["cordova", "plugin", "rm", "wallet-cordova-plugin-tests"], cwd=app_dir
+        )
+
+    subprocess.check_call(
+        ["cordova", "plugin", "add", str(tests_directory)], cwd=app_dir
+    )
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Create test harness")
+
+    platform_choices = ["android", "ios"]
+
+    parser.add_argument(
+        "--platform", required=True, nargs="+", choices=platform_choices
+    )
+    parser.add_argument("command", choices=["full", "reload-plugin", "reload-tests"])
+    parser.add_argument("-d", "--directory", type=Path)
+    parser.add_argument("-r", "--run", choices=platform_choices)
+
+    parser.add_argument("--cargo-build", dest="cargo_build", action="store_true")
+    parser.add_argument("--no-cargo-build", dest="cargo_build", action="store_false")
+    parser.set_defaults(feature=True)
+
+    args = parser.parse_args()
+
+    android = "android" in args.platform
+    ios = "ios" in args.platform
+
+    build_dir = args.directory
+
+    app_dir = build_dir / "hello"
+
+    if args.command == "full":
+        create_hello_world(build_dir)
+        install_platforms(app_dir, android=android, ios=ios)
+        install_test_framework(app_dir)
+        install_main_plugin(
+            app_dir,
+            reinstall=False,
+            android=android,
+            ios=ios,
+            cargo_build=args.cargo_build,
+        )
+        install_test_plugin(app_dir, reinstall=False)
+
+    if args.command == "reload-plugin":
+        install_main_plugin(app_dir, reinstall=True, android=android, ios=ios)
+
+    if args.command == "reload-tests":
+        install_test_plugin(app_dir, reinstall=True)
+
+    subprocess.check_call(["cordova", "build"], cwd=app_dir)
+    if args.run:
+        subprocess.check_call(["cordova", "run", args.run], cwd=app_dir)
```

Comment:
I doubt about that part. I do not know your process, but for iOS I would often want to open the project in XCode and run it from here to just have a sane debugger, or run it on an actual iPhone instead of a simulator, or to configure a simulator. So I would rather leave those as separate steps not included into this script.

Reply:
> `--run` is optional anyway, so it doesn't really change anything

Sorry, I overlooked that somehow. It is fine then.